lib/Serge/Engine/Plugin/parse_yaml.pm

```diff
@@ -134,12 +141,30 @@ sub parse {
         die $error_text;
     }
 
+    if ($self->{data}->{yaml_mode} eq 'rails') {
+        my @tree_keys = keys(%$tree);
+        my $tree_count = int(@tree_keys);
+        if ($tree_count == 0) {
+            # Special case NOOP. Empty data
+        } elsif ($tree_count == 1) {
+            $tree = $tree->{$tree_keys[0]};
+        } else {
+            my $err = join(', ', @tree_keys);
+            $err = "YAML file with more than 1 key at root in yaml_mode: rails: $err";
+            $self->{errors}->{$self->{parent}->{engine}->{current_file_rel}} = $err;
+            # Or should we die?
+            warn "WARNING: $err";
+        }
+    }
+
     # Process tree recursively
 
     $self->process_node('', $tree, $callbackref, $lang);
 
     # Reconstruct YAML
 
+    $tree = {$lang => $tree} if ($lang && $self->{data}->{yaml_mode} eq 'rails');
```

Comment:
Yes, all of this makes much more sense than what I was doing in the most recent commit.

I think it’s a totally acceptable limitation that the root of a rails-formatted yaml file and the `%LANG%` in the file name must match. There may be cases where this is a hardship, but they seem like remote hypothetical cases. We can review them and figure out a path forward if they ever occur.